backend/git_real/models.py

```diff
@@ -179,19 +178,24 @@ class Meta:
     @classmethod
     def create_or_update_from_github_api_data(cls, repo, request_body):
 
-        head_commit = request_body["head_commit"]
-        head_commit_url = head_commit["url"]
+        head_commit = request_body["head_commit"] or request_body['repository']
```

Comment:
A commit is not the same thing as a repository.
A commit url is not the same as a compare url.

Please remove these changes. 

Rule number 1 of naming things: Call it what it is. We dont call something a head commit if it is a repository.